src/Hazelcast.Net/Sql/ISqlService.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Threading.Tasks;
+
+namespace Hazelcast.Sql
+{
+    public interface ISqlService
+    {
+        Task<ISqlQueryResult> ExecuteQueryAsync(string sql, object[] parameters = null, SqlStatementOptions options = null);
```

Comment:
I am not familiar with .NET, but why this method is async? I mean, yes we make an invocation to the server, but we are not waiting for it and return an SqlResult immediately, right?

Reply:
In current implementation this will return it immediately, this is also for future compatibility in case we will need do some querying before creating `SqlResult`.

Reply:
Continued this discussion in a [new thread](https://github.com/hazelcast/hazelcast-csharp-client/pull/475/files#r675454901).